server/middleware/token.js

```diff
@@ -81,10 +77,36 @@ function token(options) {
   if (typeof currentUserLiteral === 'string') {
     currentUserLiteral = escapeRegExp(currentUserLiteral);
   }
-  assert(typeof TokenModel === 'function',
-    'loopback.token() middleware requires a AccessToken model');
 
   return function(req, res, next) {
+    var app = req.app;
+    var registry = app.registry;
+    var BaseAcessToken = registry.getModel('AccessToken');
+    var TokenModel = BaseAcessToken;
+    var defaults = [registry.getModelByType(BaseAcessToken), BaseAcessToken, loopback.AccessToken];
+
+    if(options.model) {
+      TokenModel = registry.resolveModel(options.model);
+    } else {
+      for(var i = 0; i < defaults.length; i++) {
+        TokenModel = defaults[i];
+
+        // only accept a TokenModel that has a dataSource
+        if(TokenModel && TokenModel.dataSource) {
+          break;
+        }
+      }
+    }
+
+    if(options.model) {
+      TokenModel = registry.resolveModel(options.model);
+    } else {
+      TokenModel = BaseAcessToken;
+    }
```

Comment:
I don't understand why there are two sections resolving TokenModel, one on L88-L98, the second on L101-L104. Can you please explain?

Also please cache the result of TokenModel lookup so that you resolve it only once. At the moment, you are resolving the model on every incoming request, which is IMHO premature pessimization.